asterius/src/Asterius/Types.hs

```diff
@@ -345,6 +350,21 @@ data BinaryOp
   | GeFloat64
   deriving (Show, Data)
 
+-- | 'Mutability' of variables.
+data Mutability
+  = Const  -- ^ Immutable.
```

Comment:
`Const` exists in `base`, and `Var` exists in GHC API; although the name collision is not likely, I'd still suggest just naming the datacons as `Immutable`/`Mutable` here.

Reply:
Actually I came across a collision already (in `Asterius.Foreign.DsForeign`)! I used Var/Const to match the Wasm spec but I like Mutable/Immutable better :slightly_smiling_face: